redis/src/bignum.rs

```diff
@@ -0,0 +1,74 @@
+use std::str::from_utf8;
+
+use super::{ErrorKind, RedisError, RedisResult};
+use crate::types::{FromRedisValue, RedisWrite, ToRedisArgs, Value};
+
+macro_rules! invalid_type_error {
+    ($v:expr, $det:expr) => {{
+        fail!(invalid_type_error_inner!($v, $det))
+    }};
+}
+
+macro_rules! invalid_type_error_inner {
+    ($v:expr, $det:expr) => {
+        RedisError::from((
+            ErrorKind::TypeError,
+            "Response was of incompatible type",
+            format!("{:?} (response was {:?})", $det, $v),
+        ))
+    };
+}
+
+macro_rules! from_redis_value_for_bignum_internal {
+    ($t:ty, $v:expr) => {{
+        let v = $v;
+        match *v {
+            Value::Int(val) => match <$t>::try_from(val) {
```

Comment:
you can use `map_err` instead of the matches.

Reply:
Changed to map_err